x-pack/legacy/plugins/reporting/types.d.ts

```diff
@@ -103,7 +80,27 @@ export type EnqueueJobFn = <JobParamsType>(
   request: RequestFacade
 ) => Promise<Job>;
 
-export type RequestFacade = ReportingRequest & Legacy.Request;
+export type ReportingRequestPayload = GenerateExportTypePayload | JobParamPostPayload;
+export type ReportingRequestQuery = ListQuery | GenerateQuery;
+
+export interface ReportingRequestPre {
+  management: {
+    jobTypes: any;
+  };
+  user: any;
```

Comment:
btw Security exports `AuthenticatedUser` type https://github.com/elastic/kibana/blob/02344aa29cd1284112c520e6a054b492d3849f18/x-pack/plugins/security/common/model/index.ts#L9

Reply:
I'm approve of this message

Reply:
Hm, when I tried that, I get an ESLint error:
`@kbn/eslint/no-restricted-paths: Unexpected path "../../../plugins/security/server" imported in restricted zone. Server modules cannot be imported into client modules or shared modules.`